features/cart/lib/src/bloc/cart/bloc.dart

```diff
@@ -0,0 +1,43 @@
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:domain/models/product/product_model.dart';
+
+part 'event.dart';
+
+part 'state.dart';
+
+class CartBloc extends Bloc<CartEvent, CartState> {
+  CartBloc() : super(EmptyState()) {
+    on<AddProductEvent>(_addProduct);
+    on<RemoveProductEvent>(_removeProduct);
+  }
+
+  void _addProduct(CartEvent event, Emitter emit) {
+    state.productList[event.model] = state.productList.containsKey(event.model)
```

Comment:
List<T> updatedItems = List.of(state.productList);
updated[0] = item;
...
emit(state.copyWith(productsList: updatedList));